### PR TITLE
Stanage Tensorflow docs

### PR DESCRIPTION
**Pull request by Carldkennedy**

### CODE REVIEW

File: stanagepilot/software/apps/tensorflow.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+Installation in Home Directory - GPU Version
+--------------------------------------------
+
+The GPU version of TensorFlow is a distinct Pip package and
+is also dependent on CUDA and cuDNN libraries,
+making the installation procedure slightly different.
+
+.. warning::
+   You will need to ensure you load CUDA and cuDNN modules which are compatible with the version of TensorFlow used (see :ref:`table<tensorflow_cudnn_compat_bess>`).
```
Comment (jkwmoore) [inline stanagepilot/software/apps/tensorflow.rst:76]:
```suggestion
   You will need to ensure you load CUDA and cuDNN modules which are compatible with the version of TensorFlow used (see :ref:`table<tensorflow_cudnn_compat_stanage>`).
```

File: stanagepilot/software/apps/tensorflow.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+If the wrong cuDNN module is loaded, you may receive ``ImportError`` runtime errors such as: ::
+
+   ImportError: libcublas.so.10.0: cannot open shared object file: No such file or directory
+
+This indicates that TensorFlow was expecting to find CUDA 10.0 (and an appropriate version of cuDNN) but was unable to do so.
+
+The following table shows the which module to load for the various versions of TensorFlow,
+based on the `tested build configurations <https://www.tensorflow.org/install/source#linux>`_.
+
+.. _tensorflow_cudnn_compat_bess:
```
Comment (jkwmoore) [inline stanagepilot/software/apps/tensorflow.rst:149]:
```suggestion
.. _tensorflow_cudnn_compat_stanage:
```

**Comment by Carldkennedy** [comment]
Thank you @jkwmoore

File: stanagepilot/software/apps/tensorflow.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+| 1.14.0     | 10.0 | >= 7.4 | ``cuDNN/7.4.2.24-CUDA-10.0.130``                      |
++------------+------+--------+-------------------------------------------------------+
+| 1.13.1     | 10.0 | >= 7.4 | ``cuDNN/7.4.2.24-CUDA-10.0.130``                      |
++------------+------+--------+-------------------------------------------------------+
+| >= 1.5.0   | 9.0  | 7      | N/A                                                   |
++------------+------+--------+-------------------------------------------------------+
+| >= 1.3.0   | 8.0  | 6      | N/A                                                   |
++------------+------+--------+-------------------------------------------------------+
+| >= 1.0.0   | 8.0  | 5.1    | N/A                                                   |
++------------+------+--------+-------------------------------------------------------+
```
Comment (jkwmoore) [inline stanagepilot/software/apps/tensorflow.rst:171]: I'm guessing this table isn't very cluster specific (for the SLURM clusters it should be the same iirc) so can we move this into an import, then import it into the Tensorflow pages on the SLURM clusters?

Comment (Carldkennedy) [reply stanagepilot/software/apps/tensorflow.rst]: Thank you @jkwmoore. Done as per your suggestion

File: referenceinfo/imports/software/tensorflow/compat-table-import.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+| 1.14.0     | 10.0 | >= 7.4 | ``cuDNN/7.4.2.24-CUDA-10.0.130``                                 |
++------------+------+--------+------------------------------------------------------------------+
+| 1.13.1     | 10.0 | >= 7.4 | ``cuDNN/7.4.2.24-CUDA-10.0.130``                                 |
++------------+------+--------+------------------------------------------------------------------+
+| >= 1.5.0   | 9.0  | 7      | N/A                                                              |
++------------+------+--------+------------------------------------------------------------------+
+| >= 1.3.0   | 8.0  | 6      | N/A                                                              |
++------------+------+--------+------------------------------------------------------------------+
+| >= 1.0.0   | 8.0  | 5.1    | N/A                                                              |
++------------+------+--------+------------------------------------------------------------------+
```
Comment (jkwmoore) [inline referenceinfo/imports/software/tensorflow/compat-table-import.rst:22]: Do we need to add another row for ``cuDNN/8.0.4.30-CUDA-11.1.1``? I can see this is the latest version on Bessemer. Even so, Stanage itself actually has:

```
        cuDNN/7.4.2.24-CUDA-10.0.130
        cuDNN/7.6.2.24-CUDA-10.1.243
        cuDNN/7.6.4.38-CUDA-10.0.130
        cuDNN/7.6.4.38-gcccuda-2019a
        cuDNN/7.6.4.38-gcccuda-2019b
        cuDNN/8.0.4.30-CUDA-11.1.1
        cuDNN/8.4.1.50-CUDA-11.7.0
        cuDNN/8.6.0.163-CUDA-11.8.0
        cuDNN/8.7.0.84-CUDA-11.8.0
```

I think we should cover up to the latest cudNN on both?

Comment (jkwmoore) [reply referenceinfo/imports/software/tensorflow/compat-table-import.rst:22]: Interesting, a skipped version: https://www.tensorflow.org/install/source#gpu

Marking resolved.

**Review by jkwmoore — APPROVED** [review]

**Pull request merged**